tests/python/gaia-ui-tests/gaiatest/apps/base.py

```diff
@@ -148,7 +151,15 @@ def tap_element_from_system_app(self, element=None, add_statusbar_height=False):
     def keyboard(self):
         from gaiatest.apps.keyboard.app import Keyboard
         return Keyboard(self.marionette)
+    
+    @property
+    def name(self):
+        raise NotImplementedError("{} should implement the class attribute called 'name'".format(self.__class__)  )
 
+    @property
+    def manifest_url(self):
+        app_name_to_use = self.manifest_url_app if self.manifest_url_app else self.name.lower()
```

Comment:
@mwargers remember in this case we need the entry_point that they are getting rid of it